packages/fxa-content-server/app/tests/spec/lib/router.js

```diff
@@ -513,73 +513,97 @@ describe('lib/router', () => {
   });
 
   describe('React-related methods', () => {
-    beforeEach(() => {
-      const mockAlwaysShownReactGroups = [
-        'simpleRoutes',
-        'resetPasswordRoutes',
-      ];
-      sinon
-        .stub(router, 'ALWAYS_SHOWN_REACT_GROUPS')
-        .value(mockAlwaysShownReactGroups);
-    });
     describe('showReactApp', () => {
+      var mockShowReactAppConfig;
       beforeEach(() => {
+        mockShowReactAppConfig = {
+          alwaysOnRoutes: true,
+          sometimesOnRoutes: true,
+          neverOnRoutes: false,
+          emergencyOffRoutes: false,
+          fullProdUndefinedRoutes: true,
+        };
+        const mockReactRouteGroups = {
```

Comment:
This is definitely way less flaky so nice job! 👏

I wonder if it's possible to make this more closely reflect the type that `showReactAppConfig` is, but I think the main problem with that is even if we mock these to better reflect real routes etc., if we run them through `router.js` then there's still the case of Backbone routes possibly being completely removed.

Maybe we can at least use a comment here describing why this needed to be done the way it was? Also should we modify `const showReactAppAll = {` in this file too? It's a little confusing I think with this setup and that at the top but a comment would probably mitigate that.